### PR TITLE
Added Task 3 components

### PR DESCRIPTION
**Pull request by prateekd15**
### Write Storybook stories for existing components
Create a story file for every component you created in previous modules. You should have at least one story for each of the following components (in total minimum 3):
* Counter
* SearchForm
* GenreSelect

### Implement other components
Based on [Figma prototype](https://www.figma.com/file/fKGjrOqR6nJe6LYJopGCZ8/CDP-Home-Task-%E2%80%93-React-v1?node-id=0%3A2) implement the following components:
#### MovieTile
This component should render a movie tile from the list of movies.
It should take properties to receive image url, movie name, release year, and a list of relevant genres. Alternatively, you can define one component property to take an object with all movie info. Additionally, the component should receive a callback property to capture click event.
Optionally, you can implement a context menu popup that opens when a user clicks on three dots button and contains "Edit" and "Delete" menu items.

#### MovieDetails
This component will render movie details when a movie is selected from the list (clicked). The details include movie poster image on the left and the rest of info on the right.
The component should take properties to receive image url, movie name, release year, rating, duration and a description. Alternatively,  you can specify a single property that accepts an object with all movie info.

#### SortControl
This component will render a label "Sort by" and a select control to the right. Select should have the following options:
Release Date
Title

The component should take a property that specifies current selection. Additionally, it should take a callback property to handle selection changes. The callback should be called every time a user changes "Sort by" value. The new value should be passed in callback arguments.
Use Storybook when implementing every individual component. Write a story for every component you create. It will help you to check the result and play around with some interactivity before you embed components into the app.

## Visuals

1. Default Story for SearchForm 
searchform_storybook

2. Story for SearchForm with initialValue
searchform_storybook_value

3. MovieDetails component
moviedetails

4. SortControl and MovieTile components
movietile_sortcontrol

5. Test cases
unit-tests

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,25 +3,52 @@ import './App.css';
...
   const genres = ['Action', 'Comedy', 'Drama', 'Horror', 'Science Fiction'];
+  
   const [selectedGenre, setSelectedGenre] = useState('Action');
-
   const handleGenreSelect = (genre) => {
     console.log("Selecting " + genre)
     setSelectedGenre(genre);
   };
 
+  const sortFilters = ['RELEASE DATE', 'TITLE'];
```
Comment (Ajay-Nallanagula) [inline src/App.js]: Move them to constant file , appConstants.js

File: src/component/MovieTile/MovieTile.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import './MovieTile.css';
+import MenuButton from '../MenuButton/MenuButton';
+
+function MovieTile({ imageUrl, movieName, releaseYear, genres, onSelect, id}) {
+  console.log("Started rendering movie tile component");
+  const commaSeparatedList = genres.join(', ');
+  
+  function handleOnClick() {
```
Comment (Ajay-Nallanagula) [inline src/component/MovieTile/MovieTile.js]: Suggestion: can use arrow function here , `const handleOnClick = () => onSelect(id)`

File: src/component/MovieDetails/MovieDetails.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import './MovieDetails.css';
+import searchLogo from '../../assets/search_logo.png';
+
+function MovieDetails(props) {
+    const commaSeparatedList = props.genres.join(', ');
```
Comment (Ajay-Nallanagula) [inline src/component/MovieDetails/MovieDetails.js]: You can also , destructure props

File: src/component/MovieDetails/MovieDetails.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    return (
+        <div className = 'movie-details_outer-container'>
+            <div className='movie-details_title-row'>
+                <span className='movie-details_title'>netflixroulette</span>
```
Comment (Ajay-Nallanagula) [inline src/component/MovieDetails/MovieDetails.js]: Move the string literals to constants file, this way you can use them in many places, if there is a change you can change at only one place

File: src/component/MenuButton/MenuButton.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import './MenuButton.css';
+import hoverButton from '../../assets/hover-button.png';
+
+function MenuButton() {
+
+     return (
+        <div className='movie-tile_hover-button'>
+            <img src={hoverButton} alt='Hover Button' />
+            <div className="movie-tile_menu-content">
+                <a href="#" className='movie-tile_menu-link'>Edit</a>
```
Comment (Ajay-Nallanagula) [inline src/component/MenuButton/MenuButton.js]: String literals should be moved to constant files.

**Review by Ajay-Nallanagula — COMMENTED** [review]
Most of the changes looks good. Good work!!
Few minor changes have to be done, like moving string literals.
Overall the code structure looks well organised. 

Suggestion try to use eslint.js and prettier for formatter

File: src/App.js (hunk 1 above)
Comment (prateekd15) [reply src/App.js]: Fixed as part of 5ea7447

File: src/component/MovieTile/MovieTile.js (hunk 2 above)
Comment (prateekd15) [reply src/component/MovieTile/MovieTile.js]: Fixed as part of 5ea7447

File: src/component/MovieDetails/MovieDetails.js (hunk 3 above)
Comment (prateekd15) [reply src/component/MovieDetails/MovieDetails.js]: Fixed as part of 5ea7447

File: src/component/MovieDetails/MovieDetails.js (hunk 4 above)
Comment (prateekd15) [reply src/component/MovieDetails/MovieDetails.js]: Fixed as part of 5ea7447

File: src/component/MenuButton/MenuButton.js (hunk 5 above)
Comment (prateekd15) [reply src/component/MenuButton/MenuButton.js]: Fixed as part of 5ea7447

**Comment by Ajay-Nallanagula** [comment]
All the variable names in constants should be in CAPS. Eg: EDITBUTTON,DELETEBUTTON

File: src/constants.js [hunk 6]
```diff
@@ -1,2 +1,7 @@
 export const incrementButton = "+";
-export const decrementButton = "-";
\ No newline at end of file
+export const decrementButton = "-";
+export const releaseDateLabel = "RELEASE DATE";
+export const titleLabel = "TITLE";
+export const applicationTitle = "netflixroulette";
+export const editButton = "Edit";
+export const deleteButton = "Delete";
```
Comment (Ajay-Nallanagula) [inline src/constants.js:7]: All variable names, should be in CAPS. Eg: EDITBUTTON, DELETEBUTTON

**Review by Ajay-Nallanagula — COMMENTED** [review]
Please change the variable names in constants.js to CAPS

**Review by Ajay-Nallanagula — APPROVED** [review]
Good Job!!

**Pull request merged**